### PR TITLE
[FEATURE] Add zipObject

### PR DESCRIPTION
**Pull request by skatcat31**
## Description
Add zipObject from #100 as it is one of the functions that is simple and easy to make more understandable

## What does your PR belong to?
- [ ] Website
- [x] Snippets
- [ ] General / Things regarding the repository (like CI Integration)

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking improvement of a snippet)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have checked that the changes are working properly
- [x] I have checked that there isn't any PR doing the same
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: snippets/zipObject.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+### zipObject
+
+Given an Array of valid property identifiers and an Array of values, `zipObject` returns an object associating the properties to the values
```
Comment (Pl4gue) [inline snippets/zipObject.md]: You should write "array" lowercase as in all other snippets.

File: snippets/zipObject.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+Since an object can have undefined values but not undefined property pointers, the Array of properties is used to decide the structure of the resulting object through a `reduce`
```
Comment (Pl4gue) [inline snippets/zipObject.md]: Here as well

File: snippets/zipObject.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+```js
+const zipObject = ( props, values ) => props.reduce( ( obj, prop, index ) => ( obj[prop] = values[index], obj ), {} )
+// zipObject(['a','b','c'], [1,2]) -> {a: 1, b: 2, c: undefined}
+// zipObject(['a','b'], [1,2,3]) -> {a: 1, b: 2}
+
```
Comment (Pl4gue) [inline snippets/zipObject.md]: and you may remove this empty line :smile:

**Comment by skatcat31** [comment]
@Pl4gue Are we really writing Array(capitalized because it's a specific type) lower case? huh okay I'll update

**Comment by skatcat31** [comment]
although if that were the case why didn't I do it with object...

**Pull request merged**